docs/user-guide/source-repository-hashes.md

```diff
@@ -48,3 +48,34 @@ extra-deps:
   # nix-sha256: 003lm3pm024vhbfmii7xcdd9v2rczpflxf7gdl2pyxia7p014i8z
 ```
 
+## lookupSha256
+
+In some cases we cannot modify the `cabal.project` file to add the
+`--sha256` comments. As an alternative we can pass in a `lookupSha256`
+function to get them.  For instance pandoc includes a `cabal.project`
+file in hackage includes a `source-package-reference` to `pandoc-citeproc`:
+
+```
+with builtins;
+with rec {
+  haskellNix = import (fetchTarball {
+    name   = "haskell-nix";
+    url    = https://github.com/input-output-hk/haskell.nix/archive/cb0ab3a2.tar.gz;
+    sha256 = "1ghhs2ii4ndyzlg8qjb8z3fnjsr6fd90ixn1b81lnxv40x6rkvza";
+  }) {};
+
+  pkgs = import haskellNix.sources.nixpkgs-default haskellNix.nixpkgsArgs;
+
+  pandoc = pkgs.haskell-nix.hackage-package {
+    name         = "pandoc";
+    version      = "2.9.2.1";
+    index-state  = "2020-04-15T00:00:00Z";
+    ghc          = pkgs.buildPackages.pkgs.haskell-nix.compiler.ghc865;
+    lookupSha256 = repo:
+      { "https://github.com/jgm/pandoc-citeproc"."0.17"
```

Comment:
Is this thing actually supposed to return a nested attrset like this? not just the sha256 directly?

Maybe we could make this example into a test?

Reply:
I'm asking because it looks like it just gets passed as an argument to `builtins.fetchGit`, which I thought took the sha as just a sha.

Reply:
I have added a comment and refactored the args to try to make it clearer what the code is doing.  I figured this approach would work well if you had a number of `tag` values you wanted to support. Like this:

```
lookupSha256 = { location, tag, ... }: {
  "https://github.com/jgm/pandoc-citeproc" = {
    "0.16" = "0dxx8cp2xndpw3jwiawch2dkrkp15mil7pyx7dvd810pwc22pm2q";
    "0.17" = "0dxx8cp2xndpw3jwiawch2dkrkp15mil7pyx7dvd810pwc22pm2q";
  }
}."${location}"."${tag}";
```